src/NetscriptPort.ts

```diff
@@ -8,6 +8,10 @@ const emptyPortData = "NULL PORT DATA";
 /** The object property is for typechecking and is not present at runtime */
 export type PortNumber = PositiveInteger & { __PortNumber: true };
 
+function isObject(value: unknown): value is object {
```

Comment:
I'd call this "isNonPrimitive", although it's technically true that functions are objects, it's also confusing and what we really care about here is primitive-ness.

Reply:
Oh, and because that's a sort ugly name, you could do "isPrimitive" instead and invert the logic.

Reply:
I think `isObject` more directly describes what the function is doing. I don't really understand why there would be confusion over functions, both the name (`isObject`) and return type (`value is object`) make it very clear what is going on? I could add a comment link to the mdn typeof docs or just explaining that functions are objects if that's acceptable as well?

Reply:
Also I very intentionally went with isObject instead of isPrimitive because new primitive types have been added several times and that method would break if another was added. I felt it was much safer to check for an object due to that.

Reply:
> I don't really understand why there would be confusion over functions, both the name (isObject) and return type (value is object) make it very clear what is going on?

So, when I see a function named `isObject` my first guess would be that it's checking (only) `typeof val === "object"`. Which leads to the secondary question of why even have it as a function. There's enough doubt that I will *never* be able to be certain of the exact purpose of a function named `isObject`.

> I felt it was much safer to check for an object due to that.

I wasn't suggesting that you change the way the logic works, just that you put a `!` in front of it. Or the equivalent after applying DeMorgan's law: `value === null || (typeof value !== "object" && typeof value !== "function")`

Anyway, I don't care super much, one way or another I'll merge once I get the chance to re benchmark

Reply:
> So, when I see a function named `isObject` my first guess would be that it's checking (only) `typeof val === "object"`. Which leads to the secondary question of why even have it as a function. There's enough doubt that I will *never* be able to be certain of the exact purpose of a function named `isObject`.

If the return value was just `boolean` I could see this being a concern, but it's a type predicate that states precisely what the function is doing. What room for doubt does that leave?
I also don't understand the function confusion because I never thought anything of functions being objects, you can add and modify properties so it's clearly not just an intermediary wrapper like primitives get:
```js
function sum(a, b) {
  return a + b;
}
sum.madeBy = "By: ";
console.info(sum.madeBy); // "By: "
sum.madeBy += "John Doe";
console.info(sum.madeBy); // "By: John Doe"
```

Reply:
And wait, how does your proposed change solve the problem you have? It's common knowledge that everything is either a primitive or an object, so if people think functions aren't objects they must consider them primitives? At which point they will be equally confused regardless of how the function logic is structured, because it always marks functions as objects?

Reply:
I would call it `isObjectLike` because both object and function types can be considered object-like.

In some senses, js functions are literally objects, yes. But when using `typeof` like we are, functions are classified separately from other objects, so just calling them objects is confusing.

Reply:
It's like if you had a function called "isFruit" which returned `true` for nuts. Yes, nuts are biologically fruits, but that's not how we colloquially classify them. There are multiple ways to look at them meaning of "fruit," and so *anything* called "isFruit" is ambiguous.